src/task.c

```diff
@@ -70,6 +70,11 @@ task_init_threadid(task_system *ts)
       uint64 tmp_bitmask = *tid_bitmask;
       // first bit set to 1 starting from LSB.
       uint64 pos = __builtin_ffsl(tmp_bitmask);
+
+      // If all threads are in-use, bitmask will be all 0s.
+      if (pos == 0) {
```

Comment:
@rtjohnso -- Heads-up! You may want to closely review this diff. 

I verified thru a standalone C-program that `__builtin_ffsl()` returns 0 if bitmask is 0x00. So, I think a few lines below, this is probably wrong too:

```
  78       // builtin_ffsl returns the position plus 1.
  79       tid = pos - 1;
```

The check added on L75 and bailout `goto out` seems the right thing to.

Would be nice to cross-check this subtle logic ... just in case I'm still making a mistake.